src/code/z_rumble.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * File: z_rumble.c
+ * Description: Rumble request system
+ *
+ * Provides a simple interface to allow schedulling up to RUMBLE_REQUEST_BUFFER_SIZE rumble requests to the RumblePak.
+ * There's an aditional Override type of rumble request for requests which should take priorities over any other
+ * schedulled request.
+ */
+
+#include "global.h"
+#include "z64rumble.h"
+
+RumbleManager gRumbleMgr;
+
+void Rumble_Update(void* arg0) {
+    RumbleManager_Update(&gRumbleMgr);
+    PadMgr_RumbleSet(gRumbleMgr.rumbleEnabled);
+}
+
+// Used by some bosses (and fishing)
+void Rumble_Override(f32 distSq, u8 intensity, u8 decayTimer, u8 decayStep) {
```

Comment:
Looking at how its used, I think `intensity` would be better off named `maxIntensity`, while `temp` would be the actual `intensity`.

Reply:
Or perhaps `sourceIntensity`. The idea is that the source has an inherent intensity attached to it, which is damped linearly with distance to give the received intensity.

Reply:
I like `sourceIntensity` for the parameter